pycloudlib/azure/util.py

```diff
@@ -32,20 +32,26 @@ def get_client(resource, config_dict: dict):
         The client for the resource passed as parameter.
 
     """
-    try:
-        credential = AzureCliCredential()
-        subscription_id = config_dict.get("subscriptionId")
-        client = resource(credential, subscription_id=subscription_id)
-        return client
-    except CLIError:
+    if not util.subp(["which", "az"], rcs=[0, 1]).ok:
```

Comment:
At some point `which` was going to be deprecated and printed a deprecation warning to stderr. This has been reconsidered *for now* (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=993582), so I think it's better to use `command -v` instead. Note that it requires a shell (`subp(..., shell=True`)` I think).